### ISSUE TITLE
fail gracefully when tags can not be generated for subcomponent

### ISSUE THREAD

**Issue by stagnation** [dan-t/rusty-tags#33]
I have a project with a dependency given as a github url. And no tags are generated at all.

Cargo.toml
```[package]
name = "scrape"
version = "0.1.0"
authors = ["Nils Wireklint <nils@wireklint.se>"]

[dependencies]
select = "0.3.0"

[dependencies.url]
git = "https://github.com/servo/rust-url"
```

~/a/r/scrape *master $ rusty-tags vi -v
```Fetching source of dependencies ...

Missing source 'url-52234db0c36de07437dab29c659fa9de340f18fa', might be platform specific

Creating tags ...
   for source:
      /home/spill/arbete/rust/scrape

   cached at:
      /home/spill/arbete/rust/scrape/source-temp-dir.XJGo1rQ3At6M/src_tags

Missing source 'url-52234db0c36de07437dab29c659fa9de340f18fa', might be platform specific
```

Removing the url dependency from Cargo.toml makes it possible to generate tags for the rest of the project.

----- 
Instead of crashing and not generating any tags on this kind of dependency it would be better to generate all other possible tags and emit a warning: "could not generate tags for dependency.url - (source not available|currently not supported)" or something. I am not familiar with cargo enough to tell whether the source is accessible, I would assume so but don't know. I can give you a tree listing or the full repo if you want to dig around

edit: can be reproduced on a naked new cargo project. simply add one regular semver dependency and then a github dep

**Comment by dan-t** [comment]
> ... or the full repo if you want to dig around.

Yes, please.

**Comment by stagnation** [comment]
here is the repo
https://github.com/stagnation/redditscrape-rs

forgot to tell you 
~/a/r/scrape *master $ rustc --version
```
rustc 1.16.0-nightly (4ecc85beb 2016-12-28)
```

This is with the improper url dependency.
I found out that for my needs the semver dep of url in cargo works so I have been working around the problem.

to use the semver and get it working again:
git revert 0e2bbe98897dea3665e67734a9a821942c413e43

**Comment by dan-t** [comment]
Should be fixed with `rusty-tags 1.3`.
